src/main/java/carpet/script/api/Auxiliary.java

```diff
@@ -276,6 +276,11 @@ public static void apply(Expression expression)
             ));
         });
 
+        expression.addContextFunction("item_display_name", -1, (c, t, lv)->{
+            return new FormattedTextValue(EntityValue.getItemStackFromValue(lv.get(0)).getHoverName());
+
+        });
```

Comment:
I don't fully like this function. It has too much of implementation detail.
Here's my issues (maybe not everyone sees all of them as issues, so IMO should be discussed with others too possibly on discord):

The function returns a translated text. This is an issue first of all because it's not documented in the docs (and it should maybe even be in the name? idk), but also because that implies other details and gets more error prone.

While on singleplayer it's simple: it'll translate to the client languages, it will be unexpected that on servers it will be in english (server only loads the english lang file). This is error prone, because someone may just test it in singleplayer and assume it'll always be translated, or only in multi and assume it'll never be.

Because of that, it also limits what can be done with it. Say you want to get the color of a wool block (or something, assume you can't use the id for whatever reason): You can't reliably remove the "bed" suffix from it given it'll not have the same length in all languages, and it may not even be a suffix in some languages!

At the same I also see how the translated name could be useful though, so idk, that's why I think it should be discussed. Though maybe (all of this tbd) it could be left for a more generic function to get a translated thing, since (I think?) the translated name can be gotten easily from id.

Reply:
Oh and docs should also mention more about it, like nbt having custom name will return that instead, I think they're too short for a function where details are IMO relevant.